File: TindahanNatin.Server/Models/Sale.cs
using System;

namespace TindahanNatin.Server.Models;

public class Sale
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid StoreId { get; set; }
    public Guid? ProductId { get; set; }
    public string ProductName { get; set; } = string.Empty;
    public decimal PriceAtSale { get; set; }
    public int Quantity { get; set; }
    public decimal TotalPrice { get; set; }
    public bool IsCredit { get; set; }
    public string? CustomerName { get; set; }
    public string SoldById { get; set; } = string.Empty;
    public string? SoldByName { get; set; }
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
}

File: tindahan_natin/lib/features/public_store/sale.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'sale.freezed.dart';
part 'sale.g.dart';

@freezed
abstract class Sale with _$Sale {
  const factory Sale({
    required String id,
    required String storeId,
    String? productId,
    required String productName,
    required double priceAtSale,
    required int quantity,
    required double totalPrice,
    required bool isCredit,
    String? customerName,
    required String soldById,
    String? soldByName,
    required DateTime createdAt,
  }) = _Sale;

  factory Sale.fromJson(Map<String, dynamic> json) => _$SaleFromJson(json);
}
